dictionarySearch.py

```diff
@@ -0,0 +1,27 @@
+"""Given a dictionary and a value, give list of all keys which have that value. In case of nesting, the keys should be separated by a dot.
+   {'a' : 'apple',
+     'b' : 'bobb',
+     'c' : {
+         'd' : 'dog'
+       },
+      'e' : 'dog'
+     }
+  I/P -> 'dog'
+  O/P -> ['e', 'c.d'] """
+
+d = {'a' : 'apple','b' : 'bobb','c' : {'d' : {'f':'dog'}},'e' : {'g':'dog'}}
+
+
+def value_search(value: int, dictionary: dict, matches=[], temp=""):
```

Comment:
this works, but its a python anti-pattern `matches=[]`, its best to default values to None, or have the caller pass you the container

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html